Added in the additional units from my project

Toby-Broom opened the pull request:
I added in the Osmotic Concentration, since it seemed to fit with the other quantities.  I only added the unit we needed not the whole base and derived is this acceptable?

We needed a couple more units combinations of existing units.

I added the units for the endotoxins as per the USP 85 that we are using, I didn't make a new quantity as it seems not needed as you have a count which is what this would be although you do have Microbial Formation which seems a little specialized?

I used prefix BE as E, EU are already taken.

why does the UCUM sometime have [] other times not?

fkleedorfer commented on src/main/rdf/vocab/unit/VOCAB_QUDT-UNITS-ALL-v2.1.ttl:
```diff
@@ -30526,6 +30625,19 @@ unit:MilliOHM-PER-M
   rdfs:isDefinedBy <http://qudt.org/2.1/vocab/unit> ;
   rdfs:label "milliohm per metre" .
 
+unit:MilliOSM-PER-KiloGM
```
Theoretically, the unscaled unit would be OSM per kg, so the conversionMultiplier cannot be 1 here - it must be 0.001

fkleedorfer commented:
@ralphtq @steveraysteveray @jhodgesatmb one of you needs to weigh in on the choice of 'BE' in the context of our naming convention

> I used prefix BE as E, EU are already taken.

IF `EU` is the appropriate unit name, should it not get qualifier, like `EU_something`?

Toby-Broom commented:
> @ralphtq @steveraysteveray @jhodgesatmb one of you needs to weigh in on the choice of 'BE' in the context of our naming convention
> 
> > I used prefix BE as E, EU are already taken.
> 
> IF `EU` is the appropriate unit name, should it not get qualifier, like `EU_something`?

The units are EU for the base it should be EU? It could be EndoT for example, since its a measure of the "biological endotoxin" count seemed like BE was OK with respect the term it describes.

steveraysteveray commented:
I'll think about the name, but regarding the issue of Equivalent, I finally found the last time this was discussed [here](https://github.com/qudt/qudt-public-repo/pull/882).

steveraysteveray reviewed (commented):
If you agree with my reasoning regarding equivalent quantity kinds, then your first four units would be replaced with the existing units:
BE -> NUM
BE-PER-GM -> NUM-PER-GM
BE-PER-MilliGM -> NUM-PER-MilliGM
BE-PER-MilliL -> NUM-PER-MilliL

Still thinking about OSM.
Two further notes: 
1. The definition of quantitykind:OsmoticConcentration disappeared from the PR
2. You would need to define unit:OSM before you can build the compound unit using MilliOSM.

Toby-Broom commented:
> If you agree with my reasoning regarding equivalent quantity kinds, then your first four units would be replaced with the existing units: BE -> NUM BE-PER-GM -> NUM-PER-GM BE-PER-MilliGM -> NUM-PER-MilliGM BE-PER-MilliL -> NUM-PER-MilliL
> 
> Still thinking about OSM. Two further notes:
> 
> 1. The definition of quantitykind:OsmoticConcentration disappeared from the PR
> 2. You would need to define unit:OSM before you can build the compound unit using MilliOSM.

I added it since there is CFU which is similar in that there is a NUM of microbial cells, so this is the same but endotoxins that activate the sensor, I would propose to remove CFU then as its also not needed and covered by existing as you propose.

jhodgesatmb commented:
What is BE supposed to be representing? There have been so many mails
flying around the past week that just reading all of them is taking time,
let alone commenting on them.

On Thu, Nov 14, 2024 at 12:07 PM Florian Kleedorfer <
***@***.***> wrote:

> @ralphtq <https://github.com/ralphtq> @steveraysteveray
> <https://github.com/steveraysteveray> @jhodgesatmb
> <https://github.com/jhodgesatmb> one of you needs to weigh in on the
> choice of 'BE' in the context of our naming convention
>
> I used prefix BE as E, EU are already taken.
>
> IF EU is the appropriate unit name, should it not get qualifier, like
> EU_something?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/pull/1006#issuecomment-2477310637>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWN2PDXWVCXTFA7ZJ632AT7HLAVCNFSM6AAAAABRZTQ352VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZXGMYTANRTG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

Toby-Broom commented:
I messed up this somehow that I removed the file with the changes in since there was a big refactoring and I updated my branch in the middle of it, I don't know how to fix it I think I discard this and start fresh as I don't know how to fix it.  Here was the text for the osmotic conc.

quantitykind:OsmoticConcentration
  a qudt:QuantityKind ;
  dcterms:description "The \"Osmotic Concentration\" is a quantity which characterises the solute concetration of an osmotically active compound."^^qudt:LatexString ;
  qudt:hasDimensionVector qkdv:A0E0L-3I0M1H0T0D1 ;
  qudt:informativeReference "https://en.wikipedia.org/wiki/Osmotic_concentration"^^xsd:anyURI ;
  qudt:latexDefinition "$osmolarity = \\sum_{i} \\varphi_i n_i C_i, where $\\varphi_i$ is the osmotic coefficient of the solution $n$, is the number of particles (e.g. ions), $C$  is the molar concentration of the solute, and $i$ is the index of solute."^^qudt:LatexString ;
  qudt:plainTextDescription "The \"Osmotic Concentration\" is a quantity which characterises the solute concetration of an osmotically active compound." ;
  rdfs:isDefinedBy <http://qudt.org/2.1/vocab/quantitykind> ;
  rdfs:label "Osmotic Concentration"@en .

Toby-Broom commented:
> What is BE supposed to be representing? There have been so many mails flying around the past week that just reading all of them is taking time, let alone commenting on them.
> […](#)
> On Thu, Nov 14, 2024 at 12:07 PM Florian Kleedorfer < ***@***.***> wrote: @ralphtq <https://github.com/ralphtq> @steveraysteveray <https://github.com/steveraysteveray> @jhodgesatmb <https://github.com/jhodgesatmb> one of you needs to weigh in on the choice of 'BE' in the context of our naming convention I used prefix BE as E, EU are already taken. IF EU is the appropriate unit name, should it not get qualifier, like EU_something? — Reply to this email directly, view it on GitHub <[#1006 (comment)](https://github.com/qudt/qudt-public-repo/pull/1006#issuecomment-2477310637)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AATQRWN2PDXWVCXTFA7ZJ632AT7HLAVCNFSM6AAAAABRZTQ352VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZXGMYTANRTG4> . You are receiving this because you were mentioned.Message ID: ***@***.***>
> -- Jack

BE was supposed to represent bacterial endotoxins.

jhodgesatmb commented:
That sounds like a substance and not a unit or quantity kind. You say
'was'. Was the PR removed?

On Fri, Nov 15, 2024 at 4:40 AM Toby-Broom ***@***.***> wrote:

> What is BE supposed to be representing? There have been so many mails
> flying around the past week that just reading all of them is taking time,
> let alone commenting on them.
> … <#m_3489484091480303717_>
> On Thu, Nov 14, 2024 at 12:07 PM Florian Kleedorfer < *@*.*> wrote:
> @ralphtq <https://github.com/ralphtq> https://github.com/ralphtq
> <https://github.com/ralphtq> @steveraysteveray
> <https://github.com/steveraysteveray> https://github.com/steveraysteveray
> <https://github.com/steveraysteveray> @jhodgesatmb
> <https://github.com/jhodgesatmb> https://github.com/jhodgesatmb
> <https://github.com/jhodgesatmb> one of you needs to weigh in on the choice
> of 'BE' in the context of our naming convention I used prefix BE as E, EU
> are already taken. IF EU is the appropriate unit name, should it not get
> qualifier, like EU_something? — Reply to this email directly, view it on
> GitHub <#1006 (comment)
> <https://github.com/qudt/qudt-public-repo/pull/1006#issuecomment-2477310637>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AATQRWN2PDXWVCXTFA7ZJ632AT7HLAVCNFSM6AAAAABRZTQ352VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZXGMYTANRTG4
> <https://github.com/notifications/unsubscribe-auth/AATQRWN2PDXWVCXTFA7ZJ632AT7HLAVCNFSM6AAAAABRZTQ352VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZXGMYTANRTG4>
> . You are receiving this because you were mentioned.Message ID: @.*>
> -- Jack
>
> BE was supposed to represent bacterial endotoxins.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/pull/1006#issuecomment-2478726123>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWMOWGOF3PMLFN77JGT2AXTTDAVCNFSM6AAAAABRZTQ352VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZYG4ZDMMJSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

Toby-Broom commented:
When I did the 1st PR I put 2 files in one new QuantityKind and the units that you can see in the Unit file, Unit. fkleedorfer did some changed and I took these to the fork I made then I'm not sure what I did exactly but I got myself in a mess that I don't know how to fix. I think if I close the PR and start again then it should be fine.  Sorry, I'm not so good with git.

Toby-Broom commented:
Bacterial Endotoxins are a mixture of substances, the test measures the sum of interactions of all substances to limulus factor C, so there is a count of EndoToxin interactions.  As I said to steveraysteveray I only included it since you have CFU which is also a count of material enties, so it seemed aceeptable to include.

Toby-Broom commented:
I drop this and start again as I don't know how to fix it.

Toby-Broom commented:
@steveraysteveray @jhodgesatmb @fkleedorfer I moved the discussion to #1041 , I wait until we work out what to do before adding something in.
